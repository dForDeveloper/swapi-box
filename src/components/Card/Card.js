import React from 'react';
import PropTypes from 'prop-types';

export const Card = ({ card, toggleFavorite }) => {
  const { favorite } = card;
  const displayedStats = Object.keys(card).filter(stat => {
    return stat !== 'category' && stat !== 'favorite';
  }); 
  const statLines = displayedStats.map(stat => {
    return stat === 'name' ?
      <h3 key={stat}>{card.name}</h3> :
      <p key={stat}>{stat}: {card[stat]}</p>
  });
  return (
    <div>
      {statLines}
      <button onClick={() => toggleFavorite(card)}>
        {favorite ? 'Remove from Favorites' : 'Add to Favorites'}
      </button>
    </div>
  )
}

Card.propTypes = {
  card: PropTypes.object,
  toggleFavorite: PropTypes.func
}