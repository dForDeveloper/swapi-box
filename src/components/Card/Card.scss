.Card {
  border: 2px solid white;
  border-radius: 10px;
  padding: 8px;
  display: grid;
  grid-template-rows: 80px repeat(4, 50px) 50px;

  .h3 {
    font-size: 2rem;
    text-align: center;
  }

  .span {
    display: block;
  }

  .span--key {
    font-weight: 900;
  }
  
  .span--value {
    margin-left: 20px;
    overflow-y: auto;
    height: 20px;
  }

  .button {
    border: 2px solid white;
    color: white;
    background-color: black;
    border-radius: 5px;
    height: 30px;
    padding: 0 10px;
    margin: 10px 0 0;
    outline: none;
    align-self: end;

    &:hover {
      cursor: pointer;
      border: 2px solid yellow;
      color: yellow;
    }
  }
  
  .button--fav {
    @extend .button;
    border: 2px solid yellow;
    color: yellow;

    &:hover {
      cursor: pointer;
      border: 2px solid white;
      color: white;
    }
  }

}

.Card--fav {
  @extend .Card;
  border: 2px solid yellow;
  color: yellow;
}

@media screen and (max-width: 550px) {
  .Card {
    justify-self: center;
    width: 100%;
  }
}