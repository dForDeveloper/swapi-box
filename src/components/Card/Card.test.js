import React from 'react';
import { Card } from '../Card/Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  const mockToggleFavorite = jest.fn();
  const mockPerson = {
    name: 'Luke Skywalker',
    Species: 'Human',
    Homeworld: 'some planet',
    'Population of some planet': '5 thousand',
    favorite: false,
    category: 'people'
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card
        key={mockPerson.name}
        card={mockPerson}
        toggleFavorite={mockToggleFavorite}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should call toggleFavorite when the button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockToggleFavorite).toHaveBeenCalled();
  });
});