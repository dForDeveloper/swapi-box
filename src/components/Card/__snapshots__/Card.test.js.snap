// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div>
  <h3
    key="name"
  >
    Luke Skywalker
  </h3>
  <p
    key="Species"
  >
    Species
    : 
    Human
  </p>
  <p
    key="Homeworld"
  >
    Homeworld
    : 
    some planet
  </p>
  <p
    key="Population of some planet"
  >
    Population of some planet
    : 
    5 thousand
  </p>
  <button
    onClick={[Function]}
  >
    Add to Favorites
  </button>
</div>
`;
